onnxscript/rewriter/llama_rule_sets.py

```diff
@@ -2,13 +2,73 @@
 # Licensed under the MIT License.
 from __future__ import annotations
 
+import numpy as np
+import onnx.numpy_helper
+
 import onnxscript.ir as ir
 import onnxscript.rewriter.no_op as no_op
 import onnxscript.rewriter.pattern as orp
 
 op = orp.onnxop
 
 
+class CastIdentity(orp.RewriteRuleAsClass):
+    """Replaces ``Cast(., to=to)`` by ``Identity`` if possible."""
+
+    @classmethod
+    def pattern(cls, op, x, to: int):
+        return op.Cast(x, to=to)
+
+    @classmethod
+    def rewrite(cls, op, x: ir.Value, to: int):
+        return op.Identity(x)
+
+    @classmethod
+    def check(cls, context, x, to: int) -> bool:
+        return x.dtype == to.value
+
+
+class CastCast(orp.RewriteRuleAsClass):
```

Comment:
This is not strictly the same, right? If the first cast loses precision? Eg., what happens if you cast int32 to int16 and then back to int32? Would it be correct?

Reply:
True, I can add some verifications to check it is not just about rounding. Cast + Cast is something happening a lot due to pytorch ExportedProgram and onnxscript as well.

Reply:
Is this a TODO? I don't see any checks. It would help to make sure we don't forget this.

Reply:
done, sorry, I forgot.